instance/tests/test_tasks.py

```diff
@@ -146,13 +172,29 @@ def test_make_appserver_active(self, active, mock_make_active):
         tasks.make_appserver_active(self.appserver.id, active=active)
         mock_make_active.assert_called_once_with(active=active)
 
+    @ddt.data([False, True], [False, False])
+    @patch('instance.models.server.OpenStackServer.update_status')
+    @patch('instance.models.openedx_appserver.OpenEdXAppServer.make_active')
+    def test_make_appserver_active_server_unhealthy(self, health, mock_make_active, mock_update_status):
+        """
+        By default, we activate the appserver.
+        """
+        with patch('instance.models.server.OpenStackServer.Status.Ready.is_healthy_state',
+                   new_callable=PropertyMock) as mock_is_healthy_state:
+            mock_is_healthy_state.side_effect = health
+            tasks.make_appserver_active(self.appserver.id, active=True)
+            mock_update_status.assert_called_once_with()
+            if not all(not data for data in health):
```

Comment:
This is hard to read. It's a `not` operation applied on a group of `AND`'d `not`s, so it expands to be a group of just `or`s. So this can be simplified to `if any(data for data in health)`. (I'd rename `data` to `healthy`, btw).

(For an example, see this:)
```
~(~a and ~b and ~c)
= a or b or c
```

And in the load-balancer reconfiguration context, if we see that the server is healthy _any_ time, it's good to be made active.

Reply:
Thanks for that. Really helpful.